kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/k8s/KubeClient.java

```diff
@@ -203,6 +229,25 @@ public Deployment getDeployment(String namespaceName, String deploymentName) {
         return client.apps().deployments().inNamespace(namespaceName).withName(deploymentName).get();
     }
 
+    /**
+     * Gets deployment name by prefix.
+     *
+     * @param namespaceName the namespace name
+     * @param deploymentNamePrefix the deployment name prefix
+     * @return the deployment name by prefix
+     */
+    public String getDeploymentNameByPrefix(String namespaceName, String deploymentNamePrefix) {
+        List<Deployment> prefixDeployments = client.apps().deployments().inNamespace(namespaceName).list().getItems().stream().filter(
```

Comment:
you could use [findfirst](https://www.baeldung.com/java-stream-findfirst-vs-findany)